pkg/backend/openshiftcluster/update.go

```diff
@@ -5,12 +5,29 @@ package openshiftcluster
 
 import (
 	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
 )
 
 func (m *Manager) Update(ctx context.Context) error {
-	// TODO: m.ocDynamicValidator.Dynamic is not called because it should run on
-	// an enriched oc.  Neither are we enriching oc here currently, nor does
-	// Dynamic() support running on an enriched oc.
+	var err error
+	timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+	wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+		err = m.ocDynamicValidator.Dynamic(ctx)
+		if azureerrors.HasAuthorizationFailedError(err) ||
+			azureerrors.HasLinkedAuthorizationFailedError(err) {
+			m.log.Print(err)
+			return false, nil
+		}
+		return err == nil, err
+	}, timeoutCtx.Done())
+	if err != nil {
+		return err
+	}
```

Comment:
Currently we do not have anything in user update flow and it is not invoked from anywhere. When we add features to update we can evaluate if we need this code here and whether or not we need to abort on error. At the moment the only feature I came across for the Update was SP credentials rotation, not sure what else we might needed it for.